### PR TITLE
enable wrapcheck

### PR DESCRIPTION
**Pull request by pixelsoccupied**
This should fix all the issues seen in https://github.com/openshift-kni/lifecycle-agent/pull/288 

Manually tested seedgen + all happy path stages 

/cc @donpenney

### CODE REVIEW

File: internal/precache/workload/pullImages.go [hunk 1]
```diff
@@ -98,8 +100,7 @@ func GetAuthFile() (string, error) {
 
 	// Check if authFile exists
 	if _, err := os.Stat(authFile); os.IsNotExist(err) {
-		log.Errorf("Missing auth file for podman")
-		return "", err
+		return "", fmt.Errorf("failed to get authfile for podman: %w", err)
```
Comment (browsell) [inline internal/precache/workload/pullImages.go:103]: Missing nil check

Comment (pixelsoccupied) [reply internal/precache/workload/pullImages.go:103]: L102 actually doing the init statement ...so in this case `os.IsNotExist(err)` doing the check

**Comment by pixelsoccupied** [comment]
/hold

will do another full run after the rebase

**Comment by pixelsoccupied** [comment]
/unhold

sanity check looks good

**Review by donpenney — APPROVED** [review]

**Pull request merged**